amundsen_application/static/js/components/ProfilePage/index.tsx

```diff
@@ -0,0 +1,178 @@
+import * as React from 'react';
+import * as DocumentTitle from 'react-document-title';
+import Avatar from 'react-avatar';
+import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import LoadingSpinner from '../common/LoadingSpinner';
+
+import { GlobalState } from "../../ducks/rootReducer";
+import { getUserById } from "../../ducks/user/reducer";
+import { CurrentUser, GetUserRequest } from "../../ducks/user/types";
+
+import Flag from "../common/Flag";
+import Tabs from "../common/Tabs";
+
+import './styles.scss';
+
+interface StateFromProps {
+  user: CurrentUser;
+}
+
+interface DispatchFromProps {
+  getUserById: (userId: string) => GetUserRequest;
+}
+
+type ProfilePageProps = StateFromProps & DispatchFromProps;
+
+interface ProfilePageState {
+  user: CurrentUser;
+}
+
+class ProfilePage extends React.Component<ProfilePageProps, ProfilePageState> {
+  private userId: string;
+
+  constructor(props) {
+    super(props);
+
+    const { match } = props;
+    const params = match.params;
+    this.userId = params ? params.userId : '';
+
+    this.state = {
+      user: this.props.user,
+    };
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    const { user } = nextProps;
+    return { user };
+  }
+
+  componentDidMount() {
+    this.props.getUserById(this.userId);
+  }
+
+  // TODO: consider moving logic for empty content into Tab component
+  createEmptyTabMessage = (message: string) => {
```

Comment:
I considered whether the tab component should provide a default content container or if the consuming component would rather have full control on the display. Will need to think about this some more.

Reply:
No problem. I'll leave the `TODO` in here and we can resolve it in the follow up ticket when we have backend support.